exp/controllers/awsmachinepool_controller.go

```diff
@@ -427,6 +427,13 @@ func (r *AWSMachinePoolReconciler) reconcileLaunchTemplate(machinePoolScope *sco
 		if err := ec2svc.CreateLaunchTemplateVersion(machinePoolScope, imageID, userData); err != nil {
 			return err
 		}
+		machinePoolScope.Info("start instance refresh", "number of instances", machinePoolScope.MachinePool.Spec.Replicas)
+		asgSvc := r.getASGService(ec2Scope)
+		// After creating a new version of launch template, instance refresh is required
+		// to trigger a rolling replacement of all previously launched instances.
+		if err := asgSvc.StartASGInstanceRefresh(machinePoolScope); err != nil {
+			return err
```

Comment:
Do we need to check for `InstanceRefreshInProgress` or check the status up front for the scenario here we are going around the reconciliation loop again and the refresh is still happening?

Reply:
Good point, if there is an InstanceRefreshInProgress we should block updating the template, because until an existing refresh request finalizes, cannot do new refresh request. 
Doing this change now.